core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -125,15 +125,6 @@ public int docId() {
         return this.docId;
     }
 
-    public void shardTarget(SearchShardTarget shardTarget) {
-        this.shard = shardTarget;
-        if (innerHits != null) {
-            for (InternalSearchHits searchHits : innerHits.values()) {
-                searchHits.shardTarget(shardTarget);
-            }
-        }
```

Comment:
shardTarget used to get used in inner hits, but it was removed, because the shardTarget is for all inner hits the same as the main search hit